WSRequestHandler.cpp

```diff
@@ -401,8 +413,90 @@ void WSRequestHandler::HandleStartStreaming(WSRequestHandler* req)
 {
 	if (obs_frontend_streaming_active() == false)
 	{
+		obs_data_t* streamData = obs_data_get_obj(req->data, "stream");
```

Comment:
Interestingly enough there is actually a valid use case for this.  Specifically for the case of passing metadata.  Usually metadata is specific to a stream and if that stream was stopped you would not want that metadata to pollute future streams.  If you just call SetStreamSettings, you have the risk of doing that if the stop stream event isn't received (and I found some scenarios in which it was). That's why I primarily wrote this functionality.